### PR TITLE
Microcontroller API cleanup

### PR DESCRIPTION
**Pull request by Hackerman342**

### CODE REVIEW

File: circuitpy/lib/farm_ng/utils/main_loop.py [hunk 1]
```diff
@@ -7,20 +19,17 @@
...
-from .general import TickRepeater
+from .debug_packet import FarmngDebugMemory
+from .debug_packet import FarmngDebugTimer
 from .nvm import nvm_serial_number
-from .packet import FarmngDebugMemory
-from .packet import FarmngDebugTimer
 from .packet import FarmngHeartbeat
 from .packet import NodeState
-from .packet import SupervisorReq
+from .ticks import DtTracker
```
Comment (Hackerman342) [inline circuitpy/lib/farm_ng/utils/main_loop.py:29]: Let's remove the over complication for this being able to run the display and go to the App-based approach we now use

File: circuitpy/lib/farm_ng/utils/debug_packet.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from struct import pack
+from struct import unpack
+
+from farm_ng.utils.general import clip
+from farm_ng.utils.packet import Packet
+
+
+class FarmngDebugMemory(Packet):
```
Comment (Hackerman342) [inline circuitpy/lib/farm_ng/utils/debug_packet.py:21]: Consider just removing this file when we simplify MainLoop

File: circuitpy/lib/farm_ng/utils/nvm.py [hunk 3]
```diff
@@ -1,11 +1,25 @@
+# Copyright (c) farm-ng, inc.
```
Comment (Hackerman342) [inline circuitpy/lib/farm_ng/utils/nvm.py:1]: Consider just removing this file when we simplify MainLoop

File: circuitpy/lib/farm_ng/utils/ticks.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+
+def ticks_fresh(ticks, thresh_ms=1000):
+    "Returns true if age of ticks is less than thresh_ms, assuming within 2**29 ticks of current ticks_ms()"
+    return (ticks_ms() - ticks) < thresh_ms
+
+
+def ticks_diff(ticks1, ticks2):
+    """
+    TODO: Remove me!
```
Comment (Hackerman342) [inline circuitpy/lib/farm_ng/utils/ticks.py:29]: Remove this method when we simplify MainLoop

File: circuitpy/lib/farm_ng/utils/ticks.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        now = ticks_ms()
+        if now < self.next:
+            return False
+        self.reset()
+        return True
+
+
+class DtTracker:
+    """
+    TODO: Remove me!
```
Comment (Hackerman342) [inline circuitpy/lib/farm_ng/utils/ticks.py:58]: Remove this class when we simplify MainLoop

**Comment by Hackerman342** [comment]
@mihelich this checks the box of updating the API to reflect the current state of the dashboard with release `v0.4.0`. 

We should clean up (simplify by 10x) `main_loop.py` and change to the app-based approach we use now in our internal firmware. Thoughts on that as a follow-up task vs doing it now? It would require updating all examples here, so I think it would be better to do it when I'm on site or have someone who is on site available to test, because it is definitely a risk of breaking our examples.

**Comment by guilhermedemouraa** [comment]
> @mihelich this checks the box of updating the API to reflect the current state of the dashboard with release `v0.4.0`.
> 
> We should clean up (simplify by 10x) `main_loop.py` and change to the app-based approach we use now in our internal firmware. Thoughts on that as a follow-up task vs doing it now? It would require updating all examples here, so I think it would be better to do it when I'm on site or have someone who is on site available to test, because it is definitely a risk of breaking our examples.
> 
> We could also do it as part of introducing @guilhermedemouraa or someone else on the team to basic firmware development.

I would be down to learn some firmware dev!

**Review by guilhermedemouraa — APPROVED** [review]
LGTM

**Comment by Hackerman342** [comment]
Looking over the examples, and the over complication in MainLoop is very abstracted away from the users and the examples stay simple. We can refactor this, but it's low priority IMO. Going ahead and merging.

**Comment by mihelich** [comment]
> Looking over the examples, and the over complication in MainLoop is very abstracted away from the users and the examples stay simple. We can refactor this, but it's low priority IMO.

Agree, sounds like a good FW dev starter task but not high priority.

**Pull request merged**